### PR TITLE
feature: api client package

### PR DESCRIPTION
**Pull request by drew-harris**

### CODE REVIEW

**Comment by drew-harris** [comment]
Fields are now renamed so that 
```ts
const test = await trieve.fetch("/api/chunk", "post", {
  datasetId: "c65a2dd7-298e-48e6-ac90-e336ccbbe74f",
  trDataset: "c65a2dd7-298e-48e6-ac90-e336ccbbe74f",
  requestBody: {
    chunk_html: "hello"
  }
});
```
becomes just
```ts
const test = await trieve.fetch("/api/chunk", "post", {
  datasetId: "c65a2dd7-298e-48e6-ac90-e336ccbbe74f",
  data: {
    chunk_html: "hello"
  }
});
```

File: frontends/client/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "build": "yarn build:src && yarn type:dts",
+    "build:src": "node ./scripts/build.js",
+    "type:gen": "node ./scripts/genClient.js",
+    "type:dts": "tsc --emitDeclarationOnly --declarationMap"
+  },
+  "dependencies": {},
+  "devDependencies": {
+    "typescript": "^5.5.4",
+    "esbuild": "^0.21.5",
+    "@hey-api/openapi-ts": "^0.48.2",
```
Comment (skeptrunedev) [inline frontends/client/package.json:19]:
```suggestion
    "@hey-api/openapi-ts": "^0.52.1",
```

File: frontends/client/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "build:clean": "rm -rf dist && yarn type:gen && yarn build",
+    "build": "yarn build:src && yarn type:dts",
+    "build:src": "node ./scripts/build.js",
+    "type:gen": "node ./scripts/genClient.js",
+    "type:dts": "tsc --emitDeclarationOnly --declarationMap"
+  },
+  "dependencies": {},
+  "devDependencies": {
+    "typescript": "^5.5.4",
+    "esbuild": "^0.21.5",
```
Comment (skeptrunedev) [inline frontends/client/package.json:18]:
```suggestion
    "esbuild": "^0.23.0",
```

**Review by skeptrunedev — CHANGES_REQUESTED** [review]
While I know that it breaks the turbo monorepo pattern. This new client should go into a folder called: 

`clients`

and then its own folder called `fetch-client`. 

Idea here is that we can add clients for other languages to the same folder.

File: frontends/client/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "trieve-fetch-client",
+  "type": "module",
+  "exports": "./dist/index.js",
+  "types": "./dist/index.d.ts",
+  "version": "0.0.1",
```
Comment (skeptrunedev) [inline frontends/client/package.json:6]: This should match the OpenAPI spec version

```suggestion
  "version": "0.11.7",
```

**Review by cdxker — APPROVED** [review]
Just reviewed locally, pretty happy with this. Going to make some issues for dog fooding the sdk from here.

**Pull request merged**